src/hotspot/share/prims/jvm.cpp

```diff
@@ -1016,8 +1016,6 @@ static jclass jvm_lookup_define_class(jclass lookup, const char *name,
                                   ik->is_hidden() ? "is hidden" : "is not hidden");
     }
   }
-  assert(Reflection::is_same_class_package(lookup_k, ik),
```

Comment:
I use the Lookup of the proxy interface to define a hidden class in a dynamic module as the dynamic module has no class yet and it's defined to the class loader of the proxy interface. 

We should keep the same package check if the defined class is a normal class or a hidden nestmate class.   It only allows the lookup class be in a different package for defining a hidden non-nestmate class.   This is only internal capability.    `Lookup::defineHiddenClass` API will throw IAE if the given bytes denotes a class in a different package than the lookup class.

```
+  if ((!is_hidden || is_nestmate) && !Reflection::is_same_class_package(lookup_k, ik)) { 
+    // non-hidden class or nestmate class must be in the same package as the Lookup class
+    THROW_MSG_0(vmSymbols::java_lang_IllegalArgumentException(), "Lookup class and defined class are in different packages");
+  }
+
```